server/webapp/WEB-INF/vm/build_detail/build_detail_page.vm

```diff
@@ -78,7 +78,7 @@
               <div class="sidebar_history">
                   #parse("sidebar/_sidebar_build_list.vm")
               </div>
-              <div class="job_details_content">
+              <div class="job_details_content" data-pipeline="${presenter.pipelineName}" data-pipeline-label="${presenter.pipelineLabel}" data-stage="${presenter.stageName}" data-stage-counter="${presenter.stageCounter}" data-job="${presenter.id}" data-build="${presenter.buildName}">
```

Comment:
I could do that, but in PR 3199, I split out the split out the build observer into three observers as it was really doing 3 different things.

Those fragments are used in different capacities; TabsManager uses the components to build a UID, console log log uses it for building the status URL (which will become the websocket URL if both PRs merge), and Timer uses one component as well. It was more just planning ahead.